File: DS.cpp
// An array is a type of data structure that stores elements of the same type in a contiguous block of memory. In an array,A , of size N, each memory location has some unique index,i  (where i<=i<N ), that can be referenced as A[i] or A_i.

// Reverse an array of integers.

// Note: If you've already solved our C++ domain's Arrays Introduction challenge, you may want to skip this.

// Example

// A =[1,2,3]
// Return [3,2,1]

//straightforward approach
//Time: O(n)
//Space: O(1)
vector<int> reverseArray(vector<int> a) {
    return reverse(a.begin(),a.end());
}

//the same but code by you
//Time: O(n)
//Space: O(1)
vector<int> reverseArray(vector<int> a)
{
    int l=0,r=a.size()-1;
    while(l<r)
        swap(a[l++],a[r--]);
    return a;
}

File: DS.py
"""
An array is a type of data structure that stores elements of the same type in a contiguous block of memory. In an array, , of size , each memory location has some unique index,  (where ), that can be referenced as  or .

Reverse an array of integers.

Note: If you've already solved our C++ domain's Arrays Introduction challenge, you may want to skip this.

Example

A = [1,2,3]

Return [3,2,1]

Function Description

Complete the function reverseArray in the editor below.

reverseArray has the following parameter(s):

int A[n]: the array to reverse
Returns

int[n]: the reversed array

"""
def reverseArray(a):
    l,r = 0,len(a)-1
    while(l<r):
        a[l],a[r] = a[r],a[l]
        l+=1
        r-=1
    return a
        
